pkg/system/chtimes_nowindows.go

```diff
@@ -13,3 +16,32 @@ import (
 func setCTime(path string, ctime time.Time) error {
 	return nil
 }
+
+// ChtimesNoFollow will change access and modification time on a file,
+// without following symbol links.
+func ChtimesNoFollow(path string, atime time.Time, mtime time.Time) error {
+	unixMinTime := time.Unix(0, 0)
+	unixMaxTime := maxTime
+
+	// If the modified time is prior to the Unix Epoch, or after the
+	// end of Unix Time, os.Chtimes has undefined behavior
+	// default to Unix Epoch in this case, just in case
+
+	if atime.Before(unixMinTime) || atime.After(unixMaxTime) {
+		atime = unixMinTime
+	}
+
+	if mtime.Before(unixMinTime) || mtime.After(unixMaxTime) {
+		mtime = unixMinTime
+	}
+
+	uts := []unix.Timespec{
+		unix.NsecToTimespec(atime.UnixNano()),
+		unix.NsecToTimespec(mtime.UnixNano()),
+	}
+	err := unix.UtimesNanoAt(unix.AT_FDCWD, path, uts, unix.AT_SYMLINK_NOFOLLOW)
+	if err == unix.ENOSYS {
+		err = os.Chtimes(path, atime, mtime)
+	}
```

Comment:
I think this is the only bit that differs between this one and the `Chtimes`, correct?

Haven't looked closely yet, but wondering if we should move the above to a function that's called by both, and a parameter wether or not symlinks should be followed (which would be a "no-op" for Windows).

Again, haven't looked closely if that would look better, or if the abstraction would be "too much" so just a suggestion / thought.

Reply:
Search through the code base, there are several places where Chtimes is called; Change the signature of Chtimes would change all those codes altogether,  I still prefer to adding new API ChtimesNoFollow.....

Reply:
Ah, sorry for the confusion; no, I didn't mean to change the exported Chtimes function, but to consider extracting the bits to a non-exported function that takes the argument, and that can be called by both